src/device/mod.rs

```diff
@@ -346,6 +352,7 @@ pub struct EventListeners {
     endpoint_upgrade_event_subscriber: chan::Rx<UpgradeRequestChangeEvent>,
     stun_server_subscriber: chan::Rx<Option<StunServer>>,
     post_quantum_subscriber: chan::Rx<telio_pq::Event>,
+    downgrade_event_subscriber: chan::Rx<telio_wg::DowngradeEvent>,
```

Comment:
Would it be possible to avoid events? 

In this case, `wg_controller` which is controlling the peers communication path has direct access to wg module. Maybe we can use simple method calls instead of events, as events+statefull things are usually very error-prone.